### PR TITLE
Add clear method in mempool and ProtocolHandler (3x)

### PR DESCRIPTION
**Pull request by shargon**

### CODE REVIEW

**Comment by shargon** [comment]
I would like to do this on neo-cli 2x too for test if this solve https://github.com/neo-project/neo/issues/841

**Comment by shargon** [comment]
My proposal, create a command for clear this in neo-cli, and try if this solve the issue without restart the node

**Comment by erikzhang** [comment]
We should test if it solve the issue before merging.

**Comment by shargon** [comment]
How we can test it? They said that it's not happend in local environment

**Comment by erikzhang** [comment]
Let's test it on 2.x.

**Comment by lock9** [comment]
Isn't clearing the memory pool "a bad practice"? If the transaction is sitting in the memory pool and was already "verified", isn't this going to cause some weird effect? I know we may need it "in emergencies", but clearing the whole memory pool seem to be "a little too much". Can't we clear "fewer stuff"?
Also @shargon, Erik is right, we need this added to neo 2 only. @KickSeason  is already deploying the code to testnet, and it seems that they were able to reproduce it in their testnet.

File: neo/Ledger/MemoryPool.cs [hunk 1]
```diff
@@ -200,6 +200,21 @@ public IEnumerable<Transaction> GetSortedVerifiedTransactions()
...
         }
 
+        /// <summary>
+        /// Clear mem pool
+        /// </summary>
+        public void Clear()
+        {
+            _txRwLock.EnterWriteLock();
+
+            _sortedTransactions.Clear();
```
Comment (lock9) [inline neo/Ledger/MemoryPool.cs]: I think we should avoid clearing this collection. What do you think?

Comment (shargon) [reply neo/Ledger/MemoryPool.cs]: If you want to clean all the pool, you need to clean all the collections

**Comment by lock9** [comment]
@shargon  do you still want to implement this, this way? I noticed that there is no UT too.

**Comment by shargon** [comment]
If we will add the flush command in neo-cli, it have sense, otherwise we can close it

**Comment by lock9** [comment]
@shargon  we need a second opinion. I don't think a manual flush is a good practice. I'm afraid this can cause more harm than good. I created an issue #995, maybe this will help?

**Comment by vncoelho** [comment]
I like the manual flush for:
1. mempool
2. tasks
3. protocol
4. tasks + protocol
5. flush all

**Comment by vncoelho** [comment]
#1101 considered something similar

**Pull request closed without merging**